### PR TITLE
fix: force error fixed

### PR DESCRIPTION
**Pull request by adi6859**
# Description

handle 5xx errors in resource browser and edit workflow pages

Fixes #4482
https://github.com/devtron-labs/devtron/issues/4482
## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/cdPipeline/NewCDPipeline.tsx [hunk 1]
```diff
@@ -772,7 +772,7 @@ export default function NewCDPipeline({
 
     const setForceDeleteDialogData = (serverError) => {
         const _forceDeleteData = { ...forceDeleteData }
-        setDeleteDialog(deleteDialogType.showForceDeleteDialog)
+        setDeleteDialog(deleteDialogType.showNonCascadeDeleteDialog)
```
Comment (vivek-devtron) [inline src/components/cdPipeline/NewCDPipeline.tsx:775]: is it intentional?

File: src/components/cdPipeline/NewCDPipeline.tsx [hunk 2]
```diff
@@ -819,6 +820,13 @@ export default function NewCDPipeline({
                         getWorkflows()
                     }
                 }
+                else if(response.errors){
```
Comment (vivek-devtron) [inline src/components/cdPipeline/NewCDPipeline.tsx:823]: Why this error is coming in success?

File: src/components/cdPipeline/NewCDPipeline.tsx [hunk 3]
```diff
@@ -804,6 +804,7 @@ export default function NewCDPipeline({
                         const form = { ...formData }
                         form.clusterName = response.result.deleteResponse?.clusterName
                         setFormData(form)
+                        openDeleteModal()
```
Comment (vivek-devtron) [inline src/components/cdPipeline/NewCDPipeline.tsx:807]: Is it intentional?

File: src/components/cdPipeline/NewCDPipeline.tsx (hunk 1 above)
Comment (ritvikmahajan17) [reply src/components/cdPipeline/NewCDPipeline.tsx:775]: reverting this change

File: src/components/cdPipeline/NewCDPipeline.tsx (hunk 3 above)
Comment (ritvikmahajan17) [reply src/components/cdPipeline/NewCDPipeline.tsx:807]: yes, this is for cluster unreachable case

File: src/components/cdPipeline/NewCDPipeline.tsx (hunk 2 above)
Comment (ritvikmahajan17) [reply src/components/cdPipeline/NewCDPipeline.tsx:823]: this is intentional from BE

**Review by vivek-devtron — DISMISSED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**